simulator/func_sim/func_sim.cpp

```diff
@@ -72,6 +73,9 @@ Trap FuncSim<ISA>::run( uint64 instrs_to_run)
 
         if ( instr.trap_type() == Trap::HALT)
             return instr.trap_type();
+        if ( instr.trap_type() == Trap::SYSCALL)
+            if ( kernel.get() && !kernel->execute())
```

Comment:
Do our unit tests permit the pointer be a nullptr?

Reply:
It is a temporarily solution while some unit tests use `FuncSimAndMemory`. I'll change it in next commit.